lib/liquid/variable_lookup.rb

```diff
@@ -54,6 +54,8 @@ def evaluate(context)
         elsif @command_flags & (1 << i) != 0 && object.respond_to?(key)
           object = object.send(key).to_liquid
 
+        elsif context.strict_variables
+          raise Liquid::UndefinedVariable, "undefined variable #{key} in #{object}"
```

Comment:
Here's an example where it can be useful:
```ruby
t = Template.parse("{{a}} {{b}} {{x.a}} {{x.b}}")
t.render({'a' => 1, 'x' => { 'a' => 2}}, { strict_variables: true})
# => "1  2 "
 => [#<Liquid::UndefinedVariable: Liquid error: undefined variable b>, #<Liquid::UndefinedVariable: Liquid error: undefined variable b in {"a"=>2}>]
```
The second error message gives more information to a developer where the missing variable is. If we remove `in #{object}` part we will end up with two identical errors messages:
```ruby
# => [#<Liquid::UndefinedVariable: Liquid error: undefined variable b>, #<Liquid::UndefinedVariable: Liquid error: undefined variable b>]
```